File: src/main/java/com/stolser/javatraining/webproject/model/dao/factory/DaoFactory.java
package com.stolser.javatraining.webproject.model.dao.factory;

import com.stolser.javatraining.webproject.model.dao.invoice.InvoiceDao;
import com.stolser.javatraining.webproject.model.dao.credential.CredentialDao;
import com.stolser.javatraining.webproject.model.dao.periodical.PeriodicalDao;
import com.stolser.javatraining.webproject.model.dao.role.RoleDao;
import com.stolser.javatraining.webproject.model.dao.subscription.SubscriptionDao;
import com.stolser.javatraining.webproject.model.dao.user.UserDao;

import java.sql.Connection;

public abstract class DaoFactory {
    public abstract PeriodicalDao getPeriodicalDao(Connection conn);
    public abstract CredentialDao getCredentialDao(Connection conn);
    public abstract UserDao getUserDao(Connection conn);
    public abstract RoleDao getRoleDao(Connection conn);
    public abstract SubscriptionDao getSubscriptionDao(Connection conn);
    public abstract InvoiceDao getInvoiceDao(Connection conn);

    public static DaoFactory getMysqlDaoFactory() {
        return new MysqlDaoFactory();
    }
}

File: src/main/scala/com/stolser/javatraining/webproject/dao/DaoFactory.scala
package com.stolser.javatraining.webproject.dao

/**
	* Created by Oleg Stoliarov on 11/8/18.
	*/
trait DaoFactory {
	def periodicalDao(conn: AbstractConnection): PeriodicalDao

	def credentialDao(conn: AbstractConnection): CredentialDao

	def userDao(conn: AbstractConnection): UserDao

	def roleDao(conn: AbstractConnection): RoleDao

	def subscriptionDao(conn: AbstractConnection): SubscriptionDao

	def invoiceDao(conn: AbstractConnection): InvoiceDao
}
